include/hpp/constraints/comparison-types.hh

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2020 CNRS, Airbus SAS
+// Author: Florent Lamiraux
+// Authors: Joseph Mirabel (joseph.mirabel@laas.fr), Florent Lamiraux
+//
+// This file is part of hpp-constraints.
+// hpp-constraints is free software: you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation, either version
+// 3 of the License, or (at your option) any later version.
+//
+// hpp-constraints is distributed in the hope that it will be
+// useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+// of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// General Lesser Public License for more details.  You should have
+// received a copy of the GNU Lesser General Public License along with
+// hpp-constraints. If not, see <http://www.gnu.org/licenses/>.
+
+#ifndef HPP_CONSTRAINTS_COMOARISON_TYPES_HH
+#define HPP_CONSTRAINTS_COMOARISON_TYPES_HH
+
+#include <iostream>
+#include <vector>
+#include <hpp/constraints/fwd.hh>
+
+namespace hpp {
+  namespace constraints {
+    namespace internal {
+      /// \cond HIDDEN_SYMBOLS
+      struct ReplicateCompType {
+        ComparisonType type;
+        std::size_t n;
+
+        void push_to(ComparisonTypes_t& v) const
+        {
+          for (std::size_t i = 0; i < n; ++i) v.push_back(type);
+        }
+        // Cast to ComparisonTypes_t
+        operator ComparisonTypes_t() const
+        {
+          return ComparisonTypes_t(n, type);
+        }
+      }; // struct ReplicateCompType
+      /// \endcond
+    }
+    inline internal::ReplicateCompType operator*(const int& n,
+                                                 const ComparisonType& c)
+    {
+      internal::ReplicateCompType cts;
+      cts.type = c;
+      cts.n = (std::size_t)n;
+      return cts;
+    }
+
```

Comment:
Maybe we should allow `size_type * ComparisonType` too as this is the expected size type.

```suggestion
    inline internal::ReplicateCompType operator*(const size_type& n,
                                                 const ComparisonType& c)
    {
      return operator* (static_cast<int>(n), c);
    }
```

Reply:
Adding this operator will make the compiler fail because of ambiguous overload.

Reply:
I tried and did not get a compilation error. Nevermind, you can merge it as is when ready.